### ISSUE TITLE
How to perform cascade queries?

### ISSUE THREAD

**Issue by equinoxel** [square/retrofit#1213]
I've got a situation where I need to log in and then fetch useful data. Unfortunately, the login is via a form and the site times out pretty fast. Therefore, I'd need to perform two (or more) calls in sequence (i.e. make sure the login call finished before executing the second one). So far I came with 2 variants:

1. enqueue the second call from within the `onResponse()` of the first one
2. Use Observables with map()

Is there an advised approach for this?

PS: Also on [SO](http://stackoverflow.com/q/33240340/515948).

**Comment by JakeWharton** [comment]
Questions on StackOverflow do not need duplicated here. The issue tracker is for bug reports and feature requests. I will try to answer on the linked question later (if no one has beaten me to it).

**Comment by equinoxel** [comment]
True. Thought it could have some relevance to documentation as an interesting use case.

**Comment by JakeWharton** [comment]
Fair. I need to start tracking documentation improvements as issues.

**Comment by jaumard** [comment]
Interesting is this too, with parallel requests also, it will be nice to have something included in retrofit for this :)

For now for multiple cascading request I use an asynTask and make all retrofit call synchronous in it. 
For two request you can do it in the `OnResponse` method but more will be ugly.

**Comment by JakeWharton** [comment]
The gist is that you use RxJava or ListenableFuture or CompletableFuture as
your mechanism of composition and control over the parallel/serial
execution. Retrofit itself won't be providing higher-level APIs for this,
but that also doesn't prevent you from executing Calls in parallel or
serially either, there's just a bit of manual bookkeeping.

**Comment by jiangecho** [comment]
@JakeWharton  Sorry to trouble you. And I am facing the same situation, I need to make two serial requests. You mentioned we can use RxJava or ListenableFuture or CompletableFuture to implement this mechanism. But I still can not figure out the solution, could you kindly give us a sample.
